### PR TITLE
Parse all gamemodes

### PR DESCRIPTION
**Pull request by tybug**
Adds support for parsing other gamemodes, following the reference left by @Eve-ning here: https://github.com/kszlim/osu-replay-parser/pull/27#issuecomment-845679072. This also adds ancillary `Key` enums for each respective mode.

other changes:

* rename `time_since_previous_action` to `time_delta`
* rename `keys_pressed` to `keys`
* disable rng seed warning

This will cause major breaking changes all around, so this will be a 5.0.0 release.

### CODE REVIEW

**Comment by tybug** [comment]
@Eve-ning can you check that this pr properly addresses your needs for mania parsing?

File: osrparse/utils.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+class KeyMania(IntFlag):
+    K1 = 1 << 0
+    K2 = 1 << 1
+    K3 = 1 << 2
+    K4 = 1 << 3
+    K5 = 1 << 4
+    K6 = 1 << 5
+    K7 = 1 << 6
+    K8 = 1 << 7
+
```
Comment (Eve-ning) [inline osrparse/utils.py:66]:
```suggestion
    K9 = 1 << 8
    K10 = 1 << 9
    K11 = 1 << 10
    K12 = 1 << 11
    K13 = 1 << 12
    K14 = 1 << 13
    K15 = 1 << 14
    K16 = 1 << 15
    K17 = 1 << 16
    K18 = 1 << 17
```

Comment (Eve-ning) [reply osrparse/utils.py:66]:
```
p = parse_replay_file("18KMap.osr")
data = p.play_data
np.log2(np.unique([i.x for i in data]))
```

```
>>>array([-inf,   0.,   1.,   2.,   3.,   4.,   5.,   6.,   7.,   8.,   9.,
        10.,  11.,  12.,  13.,  14.,  15.,  16.,  17.])
```

inf is from the 0

**Comment by Eve-ning** [comment]
I think that should be it, I can't really switch my fork right now without breaking a lot of things

**Pull request merged**